compiler/parser/Parser.y

```diff
@@ -787,19 +787,24 @@ module :: { Located (HsModule GhcPs) }
        : daml_version maybedocheader 'module' modid maybemodwarning maybeexports 'where' body
              {% fileSrcSpan >>= \ loc ->
                 ams (L loc (HsModule (Just $4) $6 (fst $ snd $8)
-                              (snd $ snd $8) $5 $2)
+                              (snd $ snd $8) $5 ($1 $2))
                     )
                     ([mj AnnModule $3, mj AnnWhere $7] ++ fst $8) }
         | daml_version body2
                 {% fileSrcSpan >>= \ loc ->
                    ams (L loc (HsModule Nothing Nothing
-                               (fst $ snd $2) (snd $ snd $2) Nothing Nothing))
+                               (fst $ snd $2) (snd $ snd $2) Nothing ($1 Nothing)))
                        (fst $2) }
 
-daml_version :: { () }
-  : daml version     { () }
-  | {- empty  -}     { () }
-
+daml_version :: { Maybe LHsDocString -> Maybe LHsDocString }
+  : daml version     {% do
+                          loc <- fileSrcSpan
+                          let doc = mkHsDocString "HAS_DAML_VERSION_HEADER"
```

Comment:
Does this work in haddock parse mode as well as in `Opt_KeepRawTokenStream` mode? We currently use the latter in the default `daml build` and in the IDE and the former in `daml docs`. Does it have to? I.e., what do you want to use this for?

Reply:
I don't think it matters what mode you are parsing/lexing in, because we're not using the actual haddock header, we're just using the field in the HsModule structure to store this tag. But I could be wrong?

The reason to have this to add a deprecation warning in daml-preprocessor, so we do want it to work regardless.

Reply:
I don’t know if it works or not tbh. The combination of `mkHsDocString` and the tests in the daml PR using haddock mode at least seems like it could be that it only works in haddock mode and we didn’t notice.

Reply:
Ok, I will investigate more next week, so we can be sure it works before we try to merge this.

Reply:
Added a warning in the preprocessor, and I'm happy to report that the warning is there in integration tests, daml build, and even daml studio. :)